src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/ProjectSystem/RemoteDocumentContext.cs

```diff
@@ -6,12 +6,14 @@
 
 namespace Microsoft.CodeAnalysis.Remote.Razor.ProjectSystem;
 
-internal class RemoteDocumentContext : DocumentContext
+internal sealed class RemoteDocumentContext : DocumentContext
 {
     public TextDocument TextDocument => Snapshot.TextDocument;
 
     public new RemoteDocumentSnapshot Snapshot => (RemoteDocumentSnapshot)base.Snapshot;
 
+    public IProjectQueryService ProjectQueryService => Snapshot.ProjectSnapshot.SolutionSnapshot;
```

Comment:
Nit/bikeshed: I think there is a reasonable argument to rename this to just `SolutionSnapshot`, and `IProjectQueryService` to `ISolutionSnapshot`. Yes it's a little weird that `ISolutionSnapshot` would be in a MEF composition in the LSP server, but personally I'd rather the weird things be in the code that will (hopefully!) be deleted eventually.

Reply:
I would rather change that later. I get that you dislike the word "service" because it operates on a snapshot in co-hosting, I _deeply_ dislike using the word "Snapshot" because the implementation in the non-co-hosting world will be operating a fully mutating `IProjectSnapshotManager`. IMO, using the word "snapshot" is _more_ misleading than the word "service".

Reply:
I will think further on a better name though. I never really liked `IProjectCollectionResolver` because it was oddly specific and didn't really resolve anything. Perhaps it would be better to just get away from "services" and "snapshots" and go with something like `ISolutionOperations`?

Reply:
I went forward with `ISolutionQueryOperations` and chose not to add it to the language server's DI.